### PR TITLE
k8gb extensions for metav1ext

### PR DESCRIPTION
**Pull request by kuritka**
refactor annotations into `github.com/AbsaOSS/k8gb/controllers/internal/utils`.
 - MergeAnnotations
 - ContainAnnotations

### CODE REVIEW

File: controllers/ingress.go [hunk 1]
```diff
@@ -4,9 +4,9 @@ import (
 	"context"
 	"reflect"
 
-	"github.com/AbsaOSS/k8gb/controllers/internal/utils"
-
 	k8gbv1beta1 "github.com/AbsaOSS/k8gb/api/v1beta1"
+	metav1ext "github.com/AbsaOSS/k8gb/controllers/internal/meta/v1"
```
Comment (ytsarev) [inline controllers/ingress.go:8]: I am a bit concerned to use clashing name with the upstream even if it is internal. 
@kuritka what is the motivation to do so?

**Review by ytsarev — COMMENTED** [review]
At wich point do you plan to move it out to external package?

File: controllers/ingress.go (hunk 1 above)
Comment (kuritka) [reply controllers/ingress.go]: The original idea was to have a same structure as `k8s.io/apimachinery/pkg/apis/meta/v1` which contains annotation helpers. I moved back to utils and amended changes.

**Comment by kuritka** [comment]
> At which point do you plan to move it out to external package?
I wanted to make separate PR for that. First, move utils to gopkg, then update k8gb

File: controllers/ingress.go (hunk 1 above)
Comment (somaritane) [reply controllers/ingress.go]: @kuritka can you please amend commit and PR message, so it reflect renaming to `internal/utils/annotations"?

**Review by ytsarev — CHANGES_REQUESTED** [review]
@kuritka can we please update PR header and commit message reflecting the current state of things as it is staying under `utils` ?

**Comment by kuritka** [comment]
@ytsarev, done, I will move to `gopkg` soon
thx

**Comment by kuritka** [comment]
k8gb extensions for metav1ext,  amended

**Pull request closed without merging**